vars/buildRpm.groovy

```diff
@@ -65,10 +66,15 @@ void call(Map config = [:]) {
         error_stage_result = 'UNSTABLE'
         error_build_result = 'SUCCESS'
     }
+    Map runData = ['result': 'FAILURE']
     catchError(stageResult: error_stage_result,
                buildResult: error_build_result) {
         // flow_name used as the label for this step to allow log lookup.
         sh(label: config.get('flow_name', env.STAGE_NAME),
            script: "${env_vars} " + build_script)
+        runData['result'] = 'SUCCESS'
```

Comment:
If I'm understanding correctly, we are setting a result of `SUCCESS` here even if the stage failed, but was asked to convert to `UNSTABLE`, yes?

If so, I'm not convinced that this is the correct thing to do.  While it's what we need to do to satisfy the developer requirement that package build failures don't stop the pipeline (which I don't think I agree with since a failure to build the EL8 package for example tanks all of the functional testing anyway), it seems like it's unnecessarily masking the stage's real result in our "reliability" data.

IOW, we are tainting the status to satisfy a functional requirement.  Should we also be tainting it for statistical purposes?

It might be useful for example, while looking at the data to know that the EL8 RPM build failed when we see EL8 functional stages failing.  That in fact convinces me further that we should be reporting the real stage result here, not what it had to be to satisfy the functional requirements.

Reply:
Ah yes.  I was forgetting that processing in the `catchError()` would end on any step that throws an error.  Thanks for the explanation.